src/components/Personalization/createViewChangeHandler.js

```diff
@@ -0,0 +1,32 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+export default ({ executeCachedViewDecisions, viewCache, showContainers }) => {
+  return ({ personalization, onResponse, onRequestFailure }) => {
+    if (personalization.isRenderDecisions()) {
+      executeCachedViewDecisions({
+        viewName: personalization.getViewName()
+      });
+      return;
+    }
+
+    onResponse(() => {
+      return {
+        decisions: viewCache.getView(personalization.getViewName())
```

Comment:
This is inefficient for customers that want to render their own decisions. They have to wait until the request returns before being able to get the cached decisions. I don't know how many people would want the decisions sooner, but it is something to think about for future improvements.

Reply:
I agree! but since we log an impression in Target when we return view decisions we can't just ignore them, we should return them. As a future improvement, I think that having a `render` command that will render decisions (similar to `applyOffers`) will solve this issue.